Rename two_qubit_matrix related functions

tonybruguier opened the pull request:
This could be the first step of solving issue #5041

tanujkhattar commented on cirq-core/cirq/transformers/analytical_decompositions/two_qubit_to_cz.py:
```diff
@@ -31,13 +32,25 @@
     import cirq
 
 
+@_compat.deprecated(fix='Please use two_qubit_matrix_to_cz_operations', deadline='v0.15')
```
```suggestion
@_compat.deprecated(fix='Please use cirq.two_qubit_matrix_to_cz_operations', deadline='v0.15')
```

tanujkhattar commented on cirq-core/cirq/transformers/analytical_decompositions/two_qubit_to_cz.py:
```diff
@@ -61,13 +74,29 @@ def two_qubit_matrix_to_operations(
 
 
+@_compat.deprecated(
+    fix='Please use two_qubit_matrix_to_diagonal_and_cz_operations', deadline='v0.15'
```
```suggestion
    fix='Please use cirq.two_qubit_matrix_to_diagonal_and_cz_operations', deadline='v0.15'
```

tanujkhattar commented on cirq-core/cirq/transformers/analytical_decompositions/two_qubit_to_cz.py:
```diff
@@ -31,13 +32,25 @@
     atol: float = 1e-8,
     clean_operations: bool = True,
+) -> List[ops.Operation]:
+    return two_qubit_matrix_to_cz_operations(q0, q1, mat, allow_partial_czs, atol, clean_operations)
```
nit: Don't remove the docstring on the original deprecated method, here and below.

tonybruguier replied:
Done.

tanujkhattar reviewed (approved):
LGTM % nits. 

Feel free to submit after fixing the nits.

tonybruguier commented:
Thanks for the quick turn-around on the review, @tanujkhattar 

I addressed your comments, I think, and it looks like it could be good to go, but I don't have write permissions, so I think you would have to merge it (if you think it's OK to do so, of course).
